lib/components/user/notification-prefs-pane.js

```diff
@@ -28,85 +36,393 @@ const allowedNotificationChannels = [
+      <>
+        {isEditing
+          ? (
+            // FIXME: If removing the Save/Cancel buttons on the account screen,
+            // make this <FormGroup> a <form> and remove onKeyDown handler.
+            <FormGroup validationState={phoneErrorState}>
+              <ControlLabel>Enter your phone number for SMS notifications:</ControlLabel>
+              <ControlStrip>
+                <InlinePhoneInput
+                  className='form-control'
+                  country={phoneFormatOptions.countryCode}
+                  onChange={this._handleNewPhoneNumberChange}
+                  onKeyDown={this._handlePhoneNumberKeyDown}
+                  placeholder='Enter phone number'
+                  readOnly={!isEditing}
+                  type='tel'
+                  value={newPhoneNumber}
+                />
+                <Button
+                  bsStyle='primary'
+                  disabled={isPhoneInvalid}
+                  onClick={this._handleRequestCode}
+                >
+                  Send verification text
+                </Button>
+                { // Show cancel button only if a phone number is already recorded.
+                  currentNumber && <Button onClick={this._handleCancelEditNumber}>Cancel</Button>}
+                {showPhoneError && <HelpBlock>{INVALID_PHONE_MSG}</HelpBlock>}
+              </ControlStrip>
+            </FormGroup>
+          ) : (
+            <FormGroup>
+              <ControlLabel>SMS notifications will be sent to:</ControlLabel>
+              <ControlStrip>
+                <InlineStatic>
+                  {isPending
+                    // eslint-disable-next-line jsx-a11y/label-has-for
+                    ? <>{formatPhoneNumber(newPhoneNumber)} <Label bsStyle='warning'>Pending</Label></>
+                    // eslint-disable-next-line jsx-a11y/label-has-for
+                    : <>{formatPhoneNumber(currentNumber)} {currentNumberVerified && <Label bsStyle='success'>Verified</Label>}</>
+                  }
+                </InlineStatic>
+                <Button onClick={this._handleEditNumber}>Change number</Button>
+              </ControlStrip>
+            </FormGroup>
+          )}
+
+        {isPending && (
+          // FIXME: If removing the Save/Cancel buttons on the account screen,
+          // make this <FormGroup> a <form> and remove onKeyDown handler.
+          <FormGroup validationState={codeErrorState}>
+            <p>
+              Please check the SMS messaging app on your mobile phone
+              for a text message with a verification code, and enter the code below.
+            </p>
+            <ControlLabel>Verification code:</ControlLabel>
+            <ControlStrip>
+              <Field as={InlineTextInput}
+                name='validationCode'
+                onKeyDown={this._handleValidationCodeKeyDown}
+                placeholder='123456'
+                // onBlur, onChange, and value are passed automatically by Formik
```

Comment:
Sorry, I didn't catch this earlier during testing, but this Input should have the props `type='number'` and `maxLength=6`

Reply:
On Chrome, Firefox and Safari on Mac, the browser enforces `maxLength` on `<input type='text'>` but not on `<input type='number'>.

Reply:
Putting this in a separate issue: Setting the input type to 'number' causes Formik to convert the value of the input control to a number. This means strings with leading zeros such as `001234` are deemed invalid. If we get validation codes that start with a zero this will be an issue. (I have not received validation codes that start with zero so far.)

Reply:
Correction. Twilio does send validation codes that start with zero.

Reply:
In light of the number handling above, I changed the input type back to `text` in https://github.com/opentripplanner/otp-react-redux/pull/224/commits/cbf2d86940c5b4c51d9eb03969c22926c1a2ba7b, which allows native enforcement of `maxLength`.

Reply:
@binh-dam-ibigroup, if type="number" doesn't work, let's just use `pattern="[0-9]*"` (https://stackoverflow.com/a/9318515/915811)

Reply:
The regex check is already in place at https://github.com/opentripplanner/otp-react-redux/blob/f40f4a363612c3e635c7ec593fb647eaaf8e19eb/lib/components/user/notification-prefs-pane.js#L86.

I really like to have the numeric keypad on mobile devices too. To do that, we could "cheat" and use `<input type='tel'>`, although the validation code is not really a phone number.

Reply:
Yea, if `tel` seems to work, let's try it.